### PR TITLE
Add cdf file connector

### PR DESCRIPTION
**Pull request by minshi-cognite**

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
     // The identifier of the plugin.
     // This must be globally unique, and only contain A-Za-z0-9_-
     // BEWARE: Changing this value is a dangerous operation and
     // needs manual administrator intervention to fix directory names
-    "id" : "cdf-timeseries",
+    "id" : "cdf-dataiku",
 
     // Version. It is highly recommended to use Semantic Versioning
-    "version" : "0.0.1",
+    "version" : "0.0.2",
```
Comment (sanderland) [inline plugin.json]: 0.1 please, since it's breaking

File: python-connectors/cdf-files/connector.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        Supported types are: string, int, bigint, float, double, date, boolean
+        """
+
+        # In this example, we don't specify a schema here, so DSS will infer the schema
+        # from the columns actually returned by the generate_rows method
+        return {"columns" : [ {"name": "createdTime", "type" : "string"},
+                              {"name" :"externalId", "type" : "string"},
+                              {"name" :"id", "type" : "int"},
+                              {"name" :"lastUpdatedTime", "type" : "string"},
```
Comment (sanderland) [inline python-connectors/cdf-files/connector.py]: date?

File: plugin.json (hunk 1 above)
Comment (minshi-cognite) [reply plugin.json]: Would like to use version 0.1 instead of 0.0.2?

File: python-connectors/cdf-files/connector.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                              {"name" :"metadata", "type" : "string"},
+                              {"name" :"mimeType", "type" : "string"},
+                              {"name" :"name", "type" : "string"},
+                              {"name" :"source", "type" : "string"},
+                              {"name" :"uploaded", "type" : "boolean"},
+                              {"name" :"uploadedTime", "type" : "int"}]}
```
Comment (sanderland) [inline python-connectors/cdf-files/connector.py]: also date?

File: python-connectors/cdf-files/connector.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        Supported types are: string, int, bigint, float, double, date, boolean
+        """
+
+        # In this example, we don't specify a schema here, so DSS will infer the schema
+        # from the columns actually returned by the generate_rows method
+        return {"columns" : [ {"name": "createdTime", "type" : "string"},
+                              {"name" :"externalId", "type" : "string"},
+                              {"name" :"id", "type" : "int"},
+                              {"name" :"lastUpdatedTime", "type" : "string"},
+                              {"name" :"metadata", "type" : "string"},
```
Comment (sanderland) [inline python-connectors/cdf-files/connector.py]: metadata type string is not a problem? i know their types are a bit limited

File: python-connectors/cdf-files/connector.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        Each yielded row must be a dictionary, indexed by column name.
+
+        The dataset schema and partitioning are given for information purpose.
+        """
+        files = self.client.files.list(limit=self.top_num)
+        file_dics = json.loads(str(files))
+        file_df = pd.DataFrame(file_dics)
+
+        for ix, row in file_df.iterrows():
+            yield {**{k:v for k,v in zip(file_df.columns,row)}}
```
Comment (sanderland) [inline python-connectors/cdf-files/connector.py]: rather convoluted. 
for file in files:
   yield file.dump()

here?

**Pull request merged**